bundle-size/webhooks.js

```diff
@@ -80,52 +80,74 @@ exports.installGitHubWebhooks = (app, db, githubUtils) => {
     const pullRequestId = context.payload.pull_request.number;
     const headSha = context.payload.pull_request.head.sha;
 
-    if (
-      context.payload.review.state == 'approved' &&
-      (await githubUtils.isBundleSizeApprover(approver))
-    ) {
+    if (context.payload.review.state !== 'approved') {
+      return;
+    }
+
+    const check = await getCheckFromDatabase(db, headSha);
+    if (!check) {
       context.log(
-        `Pull request ${pullRequestId} approved by a bundle-size keeper`
+        `Check ID for pull request ${pullRequestId} with head ` +
+          `SHA ${headSha} was not found in the database`
       );
+      return;
+    }
 
-      const check = await getCheckFromDatabase(db, headSha);
-      if (!check) {
-        context.log(
-          `Check ID for pull request ${pullRequestId} with head ` +
-            `SHA ${headSha} was not found in the database`
-        );
-        return;
-      }
+    const approverTeams = check.approving_teams
+      ? check.approving_teams.split(',')
+      : [];
+    if (approverTeams.length) {
+      // TODO(#617, danielrozenberg): use the result of `isBundleSizeApprover`
+      // instead of the legacy logic below.
+      const isBundleSizeApprover = githubUtils.isBundleSizeApprover(
+        approver,
+        approverTeams
+      );
+      context.log(
+        `Approving user ${approver} of pull request ${pullRequestId}`,
+        isBundleSizeApprover ? 'is' : 'is NOT',
+        'a member of',
+        approverTeams
+      );
+    }
+
+    // TODO(#617, danielrozenberg): remove the legacy logic below.
+    if (!(await githubUtils.isBundleSizeApproverLegacy(approver))) {
+      return;
+    }
 
-      let approvalMessagePrefix;
-      if (check.delta === null) {
-        context.log(
-          'Pull requests can no longer be preemptively approved for ' +
-            'bundle-size changes'
-        );
+    context.log(
+      `Pull request ${pullRequestId} approved by a bundle-size keeper`
```

Comment:
Where did "keeper" come from? :soccer:

Reply:
Old text that will be changed when we move to the new model